pkg/config/v1alpha1/types.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	configv1alpha1 "k8s.io/component-base/config/v1alpha1"
+)
+
+// ControllerConfiguration defines the desired state of GenericControllerConfiguration
+type ControllerConfiguration struct {
+	// SyncPeriod returns the SyncPeriod
+	// +optional
+	SyncPeriod *metav1.Duration `json:"syncPeriod,omitempty"`
```

Comment:
Should we document / add a default here?

Reply:
Since it's a pointer it gets set in the manager by the `setDefaultOptions()` which is why I left it blank. I think exploring apimachinery defaults at some point makes sense so when the decoding is done it can default but I don't know if we need to do that immediately.

wdyt?

Reply:
Close to what @wfernandes pointed out below, we might want to document what's defaulted at least?

Reply:
Sounds good!